app/src/main/java/com/pvh98/calculator/CalculatorFragment.kt

```diff
@@ -0,0 +1,185 @@
+package com.pvh98.calculator
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.fragment.app.Fragment
+import kotlinx.android.synthetic.main.fragment_calculator.*
+
+class CalculatorFragment : Fragment(), View.OnClickListener {
+    private var operationDecimal = false
+    private var firstValue = 0.0f
+    private var secondValue = 0.0f
+    private var operationAdd = false
+    private var operationSub = false
+    private var operationMul = false
+    private var operationDiv = false
+    private var operationEuclidDiv = false
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_calculator, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setListener()
+    }
+
+    private fun setListener() {
+        buttonZero.setOnClickListener(this)
+        buttonOne.setOnClickListener(this)
+        buttonTwo.setOnClickListener(this)
+        buttonThree.setOnClickListener(this)
+        buttonFour.setOnClickListener(this)
+        buttonFive.setOnClickListener(this)
+        buttonSix.setOnClickListener(this)
+        buttonSeven.setOnClickListener(this)
+        buttonEight.setOnClickListener(this)
+        buttonNine.setOnClickListener(this)
+        buttonDot.setOnClickListener(this)
+        buttonDelete.setOnClickListener(this)
+        buttonDeleteAll.setOnClickListener(this)
+        buttonAdd.setOnClickListener(this)
+        buttonSub.setOnClickListener(this)
+        buttonMutiple.setOnClickListener(this)
+        buttonDivide.setOnClickListener(this)
+        buttonEuclidDivision.setOnClickListener(this)
+        buttonEqual.setOnClickListener(this)
+
+    }
+
+    override fun onClick(v: View?) {
+        when (v?.id) {
```

Comment:
Tại sao cần hỏi chấm ở đây nhỉ. Hưng bt k

Reply:
Đây là Safe Call bởi nếu giá trị id lấy được là null thì nó sẽ không gây lỗi NullPointerException( một lỗi tương đối khó chịu khi làm android) . Giả sử nếu trường hợp id là null sẽ bỏ qua và không truy xuất đến id đó nữa.

Reply:
thế nếu bỏ đi thì có sao k, trong trường hợp này

Reply:
nếu bỏ đi thì nó sẽ xuất hiện dòng " Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type View? " nó sẽ bắt dùng 1 trong 3 trường hợp :
Trường hợp 1 : Ta sẽ thêm 1 câu lệnh if (v != null) bao trùm cả đoạn when 
Trường hợp 2 : Thêm toán tử "?" sau v
Trường hợp 3 : Thêm toán tử "!!" sau v thì chương trình vẫn chạy và nó có thể xuất hiện lỗi NullPointerException nếu id bị null

Reply:
Khi v!!.id mà v null thì sẽ bị NPE là rất chính xác. Vậy giá trị trả về của v?.id khi v null là gì em có biết không?

Reply:
hiểu sai ý hỏi của mình r. hàm setListener bạn để onViewCreated() thì lm sao pahri check null nữa, k phải cái j cx nên check null đâu. ? n cx như là 1 câu lệnh if...else. thêm vào thì mất thêm thời gian để check nữa. Cái này khi nào lm project sẽ chú trọng hơn. MÌnh hỏi thử xem thế nào thôi